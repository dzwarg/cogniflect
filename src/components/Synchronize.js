import React from 'react';
import StartOver from '../containers/StartOver';
import {Button, Col, Container, Jumbotron, Row} from 'react-bootstrap';
import {LinkContainer} from 'react-router-bootstrap';

const Synchronize = ({continueHandler, proceed}) => (
  <Container>
    <Row>
      <Col>
        <Jumbotron>
          <h1>Waiting for team</h1>
          <p>
            You&apos;re quick! Let&apos;s wait for the rest of your team.
          </p>
        </Jumbotron>
      </Col>
    </Row>
    <StartOver>
      <LinkContainer to="/question/1">
        {
          proceed ?
          <Button variant="success" onClick={continueHandler}>Continue</Button> :
          <Button variant="success" disabled>Continue</Button>
        }
      </LinkContainer>
    </StartOver>
  </Container>
);

export default Synchronize;
