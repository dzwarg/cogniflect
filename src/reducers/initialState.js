import {fromJS} from 'immutable';

const initialState = fromJS({
  teamCode: '',
  userTeamCode: '',
  assessmentType: 'individual',
  questionCursor: 0,
  questions: [
    {
      id: 0,
      text: "In a lake, there is a patch of lily pads. Every day, the patch doubles in size. If it takes 48 days for the patch to cover the entire lake, how long would it take for the patch to cover half of the lake?",
      truth: 1,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '24' },
        { id: 1, text: '47' },
        { id: 2, text: '96' },
        { id: 3, text: '49' }
      ],
      next: 1
    },
    {
      id: 1,
      text: "A bat and a ball cost $1.10 in total. The bat costs $1.00 more than the ball. How much does the ball cost?",
      truth: 0,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '$0.05' },
        { id: 1, text: '$0.06' },
        { id: 2, text: '$0.10' },
        { id: 3, text: '$0.50' }
      ],
      next: 2
    },
    {
      id: 2,
      text: "If it takes 5 machines 5 minutes to make 5 widgets, how long would it take 100 machines to make 100 widgets?",
      truth: 0,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '5 minutes' },
        { id: 1, text: '100 minutes' },
        { id: 2, text: '500 minutes' },
        { id: 3, text: '10000 minutes' }
      ],
      next: 3
    },
    {
      id: 3,
      text: "If you’re running a race and you pass the person in second place, what place are you in?",
      truth: 1,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '1st place' },
        { id: 1, text: '2nd place' },
        { id: 2, text: '3rd place' },
        { id: 3, text: 'Last place' }
      ],
      next: 4
    },
    {
      id: 4,
      text: "A farmer had 15 sheep and all but 8 died. How many are left?",
      truth: 2,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '6' },
        { id: 1, text: '7' },
        { id: 2, text: '8' },
        { id: 3, text: '9' }
      ],
      next: 5
    },
    {
      id: 5,
      text: "Emily’s father has three daughters. The first two are named April and May. What is the third daughter’s name?",
      truth: 3,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: 'June' },
        { id: 1, text: 'August' },
        { id: 2, text: 'Summer' },
        { id: 3, text: 'Emily' }
      ],
      next: 6
    },
    {
      id: 6,
      text: "How many cubic feet of dirt are there in a hole that is 3’ deep x 3’ wide x 3’ long?",
      truth: 0,
      myAnswer: null,
      ourAnswer: null,
      teamAnswerCount: 0,
      guesses: [
        { id: 0, text: '0 ft^3' },
        { id: 1, text: '3 ft^3' },
        { id: 2, text: '9 ft^3' },
        { id: 3, text: '27 ft^3' }
      ],
      next: null
    }
  ],
  team: {
    members: 0,
    synced: 0
  }
});

export default initialState;