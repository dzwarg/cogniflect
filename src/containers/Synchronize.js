import { connect } from 'react-redux';
import actions from '../actions';
import Synchronize from '../components/Synchronize'

const mapStateToProps = (state) => ({
  proceed: state.appState.get('assessmentType') === 'individual' &&
    state.appState.getIn(['team', 'members']) > 0 &&
    state.appState.getIn(['team', 'members']) === state.appState.getIn(['team', 'synced'])
});

const mapDispatchToProps = (dispatch) => ({
  continueHandler: (e) => {
    dispatch(actions.set(['assessmentType'], 'team'));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Synchronize);
export {mapStateToProps, mapDispatchToProps}