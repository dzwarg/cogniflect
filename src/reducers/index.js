import { combineReducers } from 'redux'
import initialState from './initialState'; 

const appState = (state = initialState, action) => {
  switch (action.type) {
    case 'SET':
      return state.setIn(action.payload.path, action.payload.value);
    default:
      return state
  }
}

export default combineReducers({
  appState
});