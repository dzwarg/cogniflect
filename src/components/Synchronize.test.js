import React from 'react';
import {SmartComponent as Synchronize, DumbComponent} from './Synchronize';
import { createStore } from 'redux';
import reducers from '../reducers';
import {fromJS} from 'immutable';
import ShallowRenderer from 'react-test-renderer/shallow';

const initialState = {
  appState: fromJS({
    assessmentType: 'individual',
    team: {
      members: 2,
      synced: 0
    }  
  })
};

const store = createStore(reducers, initialState);

describe('smart component', () => {
  var renderedComp;
  beforeEach(() => {
    const renderer = new ShallowRenderer();
    renderedComp = renderer.render(
      <Synchronize store={store}/>
    );
  });

  it('matches snapshot', () => {
    expect(renderedComp).toMatchSnapshot()
  });
  
  it('dispatches "continueHandler"', (done) => {
    store.subscribe((action) => {
      expect(store.getState().appState.get('assessmentType')).toEqual('team');
      done();
    });
    renderedComp.props.continueHandler();
  });
});

describe('dumb component', () => {
  var renderer;
  beforeEach(() => {
    renderer = new ShallowRenderer();
  });
  
  it('matches snapshot', () => {
    const renderedComp = renderer.render(
      <DumbComponent {...initialState}/>
    );
    expect(renderedComp).toMatchSnapshot();
  });
  
  it('enables "Continue"', () => {
    const enabledState = {
      appState: initialState.appState
        .setIn(['team', 'synced'], 2)
    };
    const renderedComp = renderer.render(
      <DumbComponent {...enabledState} proceed={true}/>
    );
    expect(renderedComp).toMatchSnapshot();
  });
});