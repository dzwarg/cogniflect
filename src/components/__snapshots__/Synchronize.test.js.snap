// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Synchronize enables "Continue" 1`] = `
<ForwardRef(Bootstrap(Container))>
  <ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Col))>
      <ForwardRef(Bootstrap(Jumbotron))>
        <h1>
          Waiting for team
        </h1>
        <p>
          You're quick! Let's wait for the rest of your team.
        </p>
      </ForwardRef(Bootstrap(Jumbotron))>
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
  <Memo(Connect(StartOver))>
    <withRouter(LinkContainer)
      to="/question/1"
    >
      <ForwardRef(Bootstrap(Button))
        variant="success"
      >
        Continue
      </ForwardRef(Bootstrap(Button))>
    </withRouter(LinkContainer)>
  </Memo(Connect(StartOver))>
</ForwardRef(Bootstrap(Container))>
`;

exports[`components/Synchronize matches snapshot 1`] = `
<ForwardRef(Bootstrap(Container))>
  <ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(Col))>
      <ForwardRef(Bootstrap(Jumbotron))>
        <h1>
          Waiting for team
        </h1>
        <p>
          You're quick! Let's wait for the rest of your team.
        </p>
      </ForwardRef(Bootstrap(Jumbotron))>
    </ForwardRef(Bootstrap(Col))>
  </ForwardRef(Bootstrap(Row))>
  <Memo(Connect(StartOver))>
    <withRouter(LinkContainer)
      to="/question/1"
    >
      <ForwardRef(Bootstrap(Button))
        disabled={true}
        variant="success"
      >
        Continue
      </ForwardRef(Bootstrap(Button))>
    </withRouter(LinkContainer)>
  </Memo(Connect(StartOver))>
</ForwardRef(Bootstrap(Container))>
`;
